R/formula.R

```diff
@@ -647,7 +647,8 @@ select_possible_terms_size <- function(chosen, terms, size) {
   valid_submodels <- unlist(valid_submodels[!is.na(valid_submodels)])
   if (length(chosen) > 0) {
     add_chosen <- paste0(" + ", paste(chosen, collapse = "+"))
-    remove_chosen <- paste0(" - ", paste(chosen, collapse = "-"))
+    remove_chosen <- paste0(" - ",
+                            paste(gsub("\\+", "-", chosen), collapse = "-"))
```

Comment:
As far as I can tell from playing around with the reprex from #345, this change is not necessary for fixing #345. But I'll keep it nonetheless because it comes with an improvement in terms of efficiency, as can be seen from
```r
allterms <- c("x1 + x2", "x1 + x3", "x1 + x2 + x3")
chosen <- c("x1 + x2")
size_chosen <- count_terms_chosen(chosen)
select_possible_terms_size(chosen, allterms, size = size_chosen + 1)
```
which gave
```
[1] "x3 + x2" "x2 + x3"
```
before this change and gives
```
[1] "x3"
```
after this change. Thus, the new behavior will result in fewer models to be fitted (i.e., to be projected onto) within `search_forward()`.

Reply:
And by the way, the former code just seemed like it was unintended.

Reply:
Just for my future me: Related to this `formula.R` change is the `test_varsel.R` change. The `cv_varsel.R` changes are not related to this `formula.R` change (but are indeed required to fix #345).